### PR TITLE
Add example script for ToolService with Launcher

### PR DESCRIPTION
**Pull request by nerdai**

### CODE REVIEW

File: agentfile/orchestrators/agent.py [hunk 1]
```diff
@@ -44,14 +44,12 @@ async def get_next_messages(
...
                 tools,
                 user_msg=task_def.input,
-                error_on_no_tool_call=False,
             )
         else:
             messages = memory.get()
             response = await self.llm.apredict_and_call(
                 tools_plus_human,
                 chat_history=messages,
-                error_on_no_tool_call=False,
```
Comment (logan-markewich) [inline agentfile/orchestrators/agent.py]: we should keep this line 👀 But just need the latest version of core (I updated the toml in main)

File: agentfile/launchers/local.py [hunk 2]
```diff
@@ -87,14 +101,11 @@ async def alaunch_single(self, initial_task: str) -> None:
...
-
         # runs until the message queue is stopped by the human consumer
-        await self.message_queue.start()
+        mq_task = asyncio.create_task(self.message_queue.start())
+        await asyncio.sleep(10)
+
+        # shutdown
+        for task in bg_tasks:
+            task.cancel()
+        mq_task.cancel()
```
Comment (logan-markewich) [inline agentfile/launchers/local.py]: I wonder if there's a way to return the final result here too -- maybe a future improvement

File: agentfile/orchestrators/agent.py (hunk 1 above)
Comment (nerdai) [reply agentfile/orchestrators/agent.py]: Yeah i was confused why that was there, but my core wasn't updated on my local machine! Will make the change 👍🏼

File: agentfile/launchers/local.py [hunk 3]
```diff
@@ -31,10 +32,12 @@ def __init__(
         services: List[BaseService],
         control_plane: BaseControlPlane,
         message_queue: SimpleMessageQueue,
+        additional_consumers: List[BaseMessageQueueConsumer] = [],
```
Comment (logan-markewich) [inline agentfile/launchers/local.py]: nit: we should avoid having lists as default params, due to pass-by-reference

File: agentfile/launchers/local.py (hunk 2 above)
Comment (nerdai) [reply agentfile/launchers/local.py]: I modified this to what I think is a bit more graceful shutdown.

File: agentfile/launchers/local.py (hunk 3 above)
Comment (nerdai) [reply agentfile/launchers/local.py]: good catch! I removed this entirely anyways though.

**Pull request merged**